### PR TITLE
Apply markdown linting only to added-or-changed mdfiles

### PR DESCRIPTION
**Pull request by nrgetik**
## Description
Previously, if an exception was made to merge a PR with failing checks on md files, this would result in all subsequent PRs failing on the same checks as linting was run across all files rather than only those changed/relevant in the branch to be merged. These changes should fix that (only for md files)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .github/workflows/markdown-link-checker.yml [hunk 1]
```diff
@@ -23,5 +23,7 @@ jobs:
       - name: markdown-link-check
         if: steps.verify-changed-files.outputs.files_changed == 'true'
         uses: gaurav-nelson/github-action-markdown-link-check@master
-#        with:
+        with:
+          base-branch: 'main'
```
Comment (Stromweld) [inline .github/workflows/markdown-link-checker.yml]: downside to this is that a user may not be using main or may still have master as the primary branch. Trying to keep it generic so anybody can reference and use the job.

**Comment by nrgetik** [comment]
@Stromweld Taking a somewhat different approach back at this effort but I don't remember how to test it. Can you take a peek and advise?

**Comment by Stromweld** [comment]
probably best way to test would be to use another repo and modify the workflow to point to your branch here. https://github.com/chef-cft/ca-prod-spl-private/blob/main/.github/workflows/ci-lint.yml#L22 Then modify a .md file and create a PR. It'll be throw away to test this PR's code.

**Comment by Stromweld** [comment]
Note that if you make changes here after an actions run in the throw away PR you have to make a change in that PR as well to trigger reload of the job config. Rerunning a job in a PR uses cached version of the workflow instead of reloading it.

**Comment by nrgetik** [comment]
@Stromweld I think this thing is ready for closer scrutiny. My testing seemed to go pretty alright

Again the goal here was to get these tools to run against only new or changed markdown files so that every check-on-PR wouldn't throw a ton of errors on previously-committed-and-merged markdown files with issues

File: .github/workflows/markdown-link-checker.yml [hunk 2]
```diff
@@ -7,21 +7,30 @@ jobs:
...
-        if: steps.verify-changed-files.outputs.files_changed == 'true'
-        uses: gaurav-nelson/github-action-markdown-link-check@master
-#        with:
-#          use-verbose-mode: "yes"
+      # - name: Check links in changed markdown files
+      #   uses: gaurav-nelson/github-action-markdown-link-check@v1
+      #   with:
+      #     base-branch: ${{ github.base_ref }}
+      #     check-modified-files-only: 'yes'
+      #     # use-verbose-mode: 'yes'
```
Comment (Stromweld) [inline .github/workflows/markdown-link-checker.yml]: if this is no longer needed I'd remove it. The red circle there is also letting you know that you don't have the end of file blank line. It expects the hidden end of file characters to be on their own line.

**Review by Stromweld — COMMENTED** [review]
nice work

**Comment by nrgetik** [comment]
@Stromweld I think we good now, but I don't know anything about these new failures getting flagged in `packer-iso-url-checker.yml`

**Review by Stromweld — APPROVED** [review]
LGTM

**Pull request merged**